impl/src/main/java/com/sun/faces/util/Util.java

```diff
@@ -1482,6 +1484,19 @@ public static BeanManager getCdiBeanManager(FacesContext facesContext) {
             if (result == null && facesContext != null) {
                 Map<String, Object> applicationMap = facesContext.getExternalContext().getApplicationMap();
                 result = (BeanManager) applicationMap.get("org.jboss.weld.environment.servlet.jakarta.enterprise.inject.spi.BeanManager");
+
+                if (result == null && applicationMap.get("org.jboss.weld.environment.servlet.enhancedListenerUsed") == Boolean.TRUE) {
+                    LOGGER.log(WARNING, "Weld skipped initialization - forcing it to reinitialize");
+                    try {
+                        ServletContext servletContext = (ServletContext) facesContext.getExternalContext().getContext();
+                        servletContext.setInitParameter("org.jboss.weld.environment.servlet.archive.isolation", "false");
+                        ServletContainerInitializer weld = (ServletContainerInitializer) Class.forName("org.jboss.weld.environment.servlet.EnhancedListener").getConstructor().newInstance();
+                        weld.onStartup(null, servletContext);
```

Comment:
@BalusC is this basically just for detection purposes? Assuming Weld bootstrap fails the first time around, re-triggering it for the exact same deployment will have the same result.
If it's detection only, perhaps we could instead use `ServletContext#setInitParameter()` on Weld side which you can then detect here?

Reply:
> re-triggering it for the exact same deployment will have the same result.

No, it will actually fix the issue. See also attached ticket.